fhir/src/main/java/org/openmrs/patinetflags/translators/impl/PatientFlagTranslatorImpl.java

```diff
@@ -0,0 +1,83 @@
+package org.openmrs.patinetflags.translators.impl;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import javax.annotation.Nonnull;
+
+import java.util.Collections;
+
+
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Flag;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Narrative;
+import org.hl7.fhir.r4.model.Period;
+import org.hl7.fhir.utilities.xhtml.XhtmlNode;
+import org.openmrs.module.fhir2.api.translators.PatientReferenceTranslator;
+import org.openmrs.module.patientflags.PatientFlag;
+import org.openmrs.patinetflags.translators.PatientFlagTranslator;
+import org.openmrs.patinetflags.translators.PriorityTranslator;
+import org.openmrs.patinetflags.translators.TagTranslator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PatientFlagTranslatorImpl implements PatientFlagTranslator {
+
+    @Autowired
+    private PatientReferenceTranslator patientReferenceTranslator;
+
+    @Autowired
+    private TagTranslator tagTranslator;
+
+    @Autowired
+    private PriorityTranslator priorityTranslator;
+
+    @Override
+    public Flag toFhirResource(@Nonnull PatientFlag patientFlag) {
+        notNull(patientFlag, "The Openmrs Patient flag object should not be null");
+
+        Flag flag = new Flag();
+
+        flag.setId(patientFlag.getUuid());
+
+        Narrative narrative = new Narrative();
+        narrative.setStatus(Narrative.NarrativeStatus.GENERATED);
+
+        XhtmlNode xhtmlNode = new XhtmlNode();
+        xhtmlNode.setName(patientFlag.getMessage());
+        narrative.setDiv(xhtmlNode);
+        flag.setText(narrative);
```

Comment:
Don't do this in a translator.

Reply:
@ibacher where should I add this ?

Reply:
It'd be best if you could plug into the narrative generation system [here](https://github.com/openmrs/openmrs-module-fhir2/tree/master/omod/src/main/java/org/openmrs/module/fhir2/narrative), otherwise, just leave it alone for now (I don't, off-hand, remember how that system works).